src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -204,7 +204,7 @@ def _is_downgrade(v1: str, v2: str) -> bool:
         Returns True if version v1 is less than version v2.
         """
         try:
-            return packaging_version.Version(v1) < packaging_version.Version(v2)
-        except packaging_version.InvalidVersion:
+            return VersionInfo.parse(v1) < VersionInfo.parse(v2)
+        except ValueError:
```

Comment:
May I ask why do you need to use `semver` instead of `packaging`?

Reply:
The reason is mentioned in [https://github.com/AzureArcForKubernetes/azure-cli-extensions/pull/241](https://github.com/AzureArcForKubernetes/azure-cli-extensions/pull/241). Please let me know if you need more information.